src/main/java/com/intel/gkl/IntelGKLUtils.java

```diff
@@ -113,27 +113,58 @@ private static void tryLoad(String resourcePath, File tempDir) throws Unsatisfie
      * @param tempDir directory where the shared library file is extracted
      * @return {@code true} if GKL loaded successfully, {@code false} otherwise
      */
-    public static synchronized boolean load(File tempDir) {
-        if (isLoaded) { return true; }
+    public static synchronized boolean load(File tempDir, String libFileName) {
 
-        if (!isAvxSupported()) { return false; }
+        if (!isAvxSupported()) {
```

Comment:
We need to remove the `isAvxSupported` check from `load` because some native libraries may not require AVX. We should check for AVX support as needed in each class.